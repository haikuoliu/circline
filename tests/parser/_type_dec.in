int a;
float a;
bool a;
node a;
graph a;
list a;
dict<int>a;
dict<float>a;
dict<string>a;
dict<node>a;
dict<graph>a;
int a = 1;
float a = 1.0;
bool a = true;
node a = node(1);
graph a = node(1)--node(2);
list a = [];
dict<int>a = {};
