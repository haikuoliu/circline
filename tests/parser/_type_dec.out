Local(int, a, Noexpression)
Local(float, a, Noexpression)
Local(bool, a, Noexpression)
Local(node, a, Noexpression)
Local(graph, a, Noexpression)
Local(list, a, Noexpression)
Local(dict<int>, a, Noexpression)
Local(dict<float>, a, Noexpression)
Local(dict<string>, a, Noexpression)
Local(dict<node>, a, Noexpression)
Local(dict<graph>, a, Noexpression)
Local(int, a, Num_Lit(1))
Local(float, a, Num_Lit(1.))
Local(bool, a, Bool_lit(true))
Local(node, a, Node(Num_Lit(1)))
Local(graph, a, Graph_Link(Node(Num_Lit(1)), DLink, Node(Num_Lit(2)), WithEdge, Null))
Local(list, a, List())
Local(dict<int>, a, Dict())
